### PR TITLE
Introduce torch.sym_add, variadic add

### PR DESCRIPTION
**Pull request by laithsakka**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138660

Tested internally here: https://www.internalfb.com/diff/D64057744
This is a reland after previous internal failures.
main change is 
```
 if min is None and max is None:
        torch._check_is_size(size)
        return
```

Partially addresses https://github.com/pytorch/pytorch/issues/128150

When you have big sums of values, we end up computing long chains of
binary addition in our FX graph representation.  Not only is this ugly,
it also is quadratic, as the sympy.Add constructor is O(N) in number
of arguments.  Instead, ensure that we maintain the summation as a
single FX node so we can do the entire addition all in one go.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @ezyang @SherlockNoMad @EikanWang @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by laithsakka** [comment]
rebase

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -1010,6 +1013,38 @@ def call_function(
...
+                    [
+                        (
+                            x.value
+                            if isinstance(x, variables.ConstantVariable)
+                            else x.sym_num
+                        )
+                        for x in args[0].items
+                    ]
+                ),
+            )
```
Comment (XuehaiPan) [inline torch/_dynamo/variables/functions.py:1046]: I'd like to create a dispatch registry to make the code easier to maintain.

```python
check_fn, dispatch_fn = self.shortcuts.get(self.fn, (None, None))
if check_fn is not None and check_fn(args, kwargs):
    return dispatch_fn(self, tx, args, kwargs)
```

**Review by bobrenjc93 — APPROVED** [review]

**Pull request closed without merging**